clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -4770,9 +4770,20 @@ std::string CompilerInvocation::getModuleHash() const {
 
   // When compiling with -gmodules, also hash -fdebug-prefix-map as it
   // affects the debug info in the PCM.
-  if (getCodeGenOpts().DebugTypeExtRefs)
+  if (getHeaderSearchOpts().ModuleFormat == "obj") {
```

Comment:
Why did this change?

Reply:
There was a test case that only passed the module format, but not DebugTypeExtRefs. DebugTypeExtRefs was used as a proxy for `-gmodules`, but that doesn't always work. From my understanding the driving factor should be the ModuleFormat, because that decides if debug options make a difference in the PCM.

Reply:
CC: @adrian-prantl to confirm my understanding